packages/dd-trace/src/encode/chunk.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const DEFAULT_MIN_SIZE = 2 * 1024 * 1024 // 2MB
+
+class Chunk {
+  constructor (minSize = DEFAULT_MIN_SIZE) {
+    this.buffer = Buffer.allocUnsafe(minSize)
+    this.length = 0
+    this._minSize = minSize
+  }
+
+  push (...bytes) {
+    this._reserve(bytes.length)
+
+    for (const byte of bytes) {
+      this.buffer[this.length++] = byte
+    }
+  }
+
+  write (value) {
+    const length = Buffer.byteLength(value)
+
+    if (length < 0x20) { // fixstr
+      this.push(length | 0xa0)
+    } else if (length < 0x100) { // str 8
+      this.push(0xd9, length)
+    } else if (length < 0x10000) { // str 16
+      this.push(0xda, length >> 8, length)
+    } else if (length < 0x100000000) { // str 32
+      this.push(0xdb, length >> 24, length >> 16, length >> 8, length)
+    } else {
+      throw new Error('String too long')
```

Comment:
Where does this error get caught?

i.e. is there some test verifying that this doesn't crash anything?

Reply:
I could just remove it since it would take a >4GB string to happen, which is very unlikely. It's only there because it was copy pasted from notepack.

Reply:
Well, even unlikely scenarios can happen, so I'd rather keep it in. Since this is part of encoding, which already is surrounded by error handling, I think it's fine to leave as-is.